scipy/stats/_distribution_infrastructure.py

```diff
@@ -502,18 +503,48 @@ class _IntegerDomain(_SimpleDomain):
         Returns a string representation of the domain, e.g. "{a, a+1, ..., b-1, b}".
 
     """
-
     def contains(self, item, parameter_values=None):
         super_contains = super().contains(item, parameter_values)
         integral = (item == np.round(item))
         return super_contains & integral
 
     def __str__(self):
         a, b = self.endpoints
-        a = self.symbols.get(a, f"{a}")
-        b = self.symbols.get(b, f"{b}")
-        ap1 = f"{a} + 1" if isinstance(a, str) else f"{a + 1}"
-        bm1 = f"{b} - 1" if isinstance(b, str) else f"{b - 1}"
+        a = self.symbols.get(a, a)
+        b = self.symbols.get(b, b)
+
+        a_str, b_str = isinstance(a, str), isinstance(b, str)
+        a_inf = a == r"-\infty" if a_str else np.isinf(a)
+        b_inf = b == r"\infty" if b_str else np.isinf(b)
+
+        # This doesn't work well for cases where ``a`` is floating point
+        # number large enough that ``nextafter(a, inf) > a + 1``, and
+        # similarly for ``b`` and nextafter(b, -inf). There may not be any
+        # distributions fit for SciPy where we would actually need to handle these
+        # cases though.
+        ap1 = f"{a} + 1" if a_str else f"{a + 1}"
+        bm1 = f"{b} - 1" if b_str else f"{b - 1}"
+
+        if not a_str and not b_str:
+            gap = b - a
+            if gap == 3:
+                return f"{{{a}, {ap1}, {bm1}, {b}}}"
```

Comment:
I think the problem (before, too) is that we need to escape the curly braces, since they mean something in LaTeX; e.g., `${}$` renders as ${}$ vs `$\\{\\}$` reenders as $\\{\\}$. Super confusing when combined with curly braces in f-strings.